solcjs

```diff
@@ -89,8 +89,17 @@ var output = solc.compile({ sources: sources }, argv.optimize ? 1 : 0);
 if (!output) {
   abort('No output from compiler');
 } else if (output['errors']) {
+  function isWarning (message) {
+    return message.match(/^(.*:[0-9]*:[0-9]* )?Warning: /)
```

Comment:
This is the regexp from browser-solidity so it is "battle tested". Eventually this entire code should move onto jsonio alleviating this problem.